designs/autoscaler.md

```diff
@@ -0,0 +1,515 @@
+}
+```
+
+### `AutoScalingConfiguration` Validation
+
+If `AutoScalingConfiguration` is not specified, we won’t add any annotation to the underlying MachineDeployment for the worker node group. Instead, we rely on the worker node group count to provision fixed number of nodes.
+
+When `AutoScalingConfiguration` is specified, we validate
+
+* minCount > 0, since Cluster Autoscaler with CAPI does not yet support scaling down to 0
+* minCount <= count <= maxCount
+
+### CAPI  Machine Deployment
+
+Internally, the `minCount` and `maxCount` defined in the EKS-A cluster spec is converted to the corresponding worker node group MachineDeployment’s annotations `cluster.x-k8s.io/cluster-api-autoscaler-node-group-max-size` and `cluster.x-k8s.io/cluster-api-autoscaler-node-group-min-size`, as EKS-A builds the CAPI templates. The Cluster Autoscaler will monitor any of those MachineDeployment containing both of these annotations.
+
+```yaml
+apiVersion: cluster.x-k8s.io/v1beta1
+kind: MachineDeployment
+metadata:
+  annotations:
+    cluster.x-k8s.io/cluster-api-autoscaler-node-group-max-size: {{.maxCount}}
+    cluster.x-k8s.io/cluster-api-autoscaler-node-group-min-size: {{.minCount}}
+    cluster.x-k8s.io/cluster-name: {{.clusterName}}
+  name: {{.workerNodeGroupName}}
+  namespace: eksa-system
+spec:
+  clusterName: {{.clusterName}}
+  replicas: {{.count}}
+  ...
+```
+
+Note: the cluster will not have any autoscaling functionality until the Cluster Autoscaler controller is properly configured and installed in the cluster. Setting up the MachineDeployment annotations only guarantees that Cluster Autoscaler can recognize the worker node group and its autoscaling configuration. We will add a warning message in CLI logs when the autoscaling is configured in EKS-A cluster spec but autoscaler is not installed in the cluster.
+
+### Cluster Autoscaler Controller
+
+Cluster Autoscaler is a tool that automatically adjusts the size of the Kubernetes cluster. Specifically, it can:
+
+* increase the size of the cluster when there are pending pods due to insufficient resources.
+* decrease the size of the cluster when some nodes have low utilization for a significant amount of time.
+
+Just like any other application, Cluster Autoscaler can be installed as a `Deployment` in any Kubernetes cluster. In order to connect Cluster Autoscaler to Cluster API managed EKS Anywhere cluster, a user needs to configure Cluster Autoscaler to use `clusterapi` as the cloud provider, and provide the path to the kubeconfig for the target cluster to autoscale.
+
+Currently, it is impossible to have one autoscaler controller to watch multiple workload clusters with different kubeconfig at the same time, as there is no way currently to separate the pods between the workload clusters autoscaler is watching. The recommended method is to run multiple autoscalers in the management cluster with different options for how they discover the scalable resource sets.
+
```

Comment:
I don't know what customers want, but it seems like it would be way easier to choose one deployment. The easiest thing I think would be run the autoscaler on the cluster it is managing which would cover both management and workload clusters.